### PR TITLE
Merge User API & Create Post method for Songs API

### PR DESCRIPTION
**Pull request by khanhphd-student**

### CODE REVIEW

File: server/controller/Songs/attribute.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const SONG_ATTRIBUTE = {
+    songID: `songID`,
+    songName: `songname`,
```
Comment (TrNgTien) [inline server/controller/Songs/attribute.js]: I have already adjust naming convention in MySQL. Please check it and test again! @Dev01-vinamr-Ken

File: server/controller/Songs/songHandler.js [hunk 2]
```diff
@@ -1,26 +1,57 @@
...
-            message: 'error',
-            error: error
-        })
-    }
+// Get all songs
+module.exports.getAllSongs = async (data, res) => {
+  try {
+    let connection = await dbConnection();
+    let getSongsQuery =
+      "SELECT songname, uploadDate, numberOfLikes, duration FROM `songs`";
```
Comment (TrNgTien) [inline server/controller/Songs/songHandler.js]: I have already adjust naming convention in MySQL. Please check it and test again! @Dev01-vinamr-Ken

File: server/controller/Songs/songHandler.js [hunk 3]
```diff
@@ -1,26 +1,57 @@
...
-module.exports.postSong = async (req, res) => {
-    //To do
-}
+// Dealing when admin post song
+module.exports.postSong = async (data, res) => {
+  try {
+    let connection = await dbConnection();
+    let postSongsQuery = `
+      INSERT INTO songs(
+        ${SONG_ATTRIBUTE.songID},
```
Comment (TrNgTien) [inline server/controller/Songs/songHandler.js]: We never add ID, when I create table songs I did like this:
songID int AUTO_INCREMENT NOT NULL,

File: server/controller/Songs/songHandler.js [hunk 4]
```diff
@@ -1,26 +1,57 @@
...
+        ${SONG_ATTRIBUTE.songName}, 
+        ${SONG_ATTRIBUTE.uploadDate}, 
+        ${SONG_ATTRIBUTE.numberOfLikes}, 
+        ${SONG_ATTRIBUTE.duration}
+      )
+      VALUES(
+        ${getRandomInt(10)},
```
Comment (TrNgTien) [inline server/controller/Songs/songHandler.js]: please delete that

File: server/controller/Songs/songHandler.js [hunk 5]
```diff
@@ -2,14 +2,18 @@ const sqlQuery = require("../../database/my_sql_query");
 const dbConnection = require("../../database/db_connection");
 
 import { SONG_ATTRIBUTE } from "./attribute";
```
Comment (TrNgTien) [inline server/controller/Songs/songHandler.js:4]: This is FE syntax :)))

File: server/controller/Songs/attribute.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export const SONG_ATTRIBUTE = {
```
Comment (TrNgTien) [inline server/controller/Songs/attribute.js:1]: please refactor that, BE uses module

File: server/controller/Songs/attribute.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    songID: `songID`,
```
Comment (TrNgTien) [inline server/controller/Songs/attribute.js:2]: please delete that!

Comment (TrNgTien) [inline server/controller/Songs/attribute.js:2]: please delete that when we don't need that

**Comment by TrNgTien** [comment]
I want your work can be tested and demo the data @Dev01-vinamr-Ken

File: server/controller/Songs/songHandler.js (hunk 5 above)
Comment (khanhphd-student) [reply server/controller/Songs/songHandler.js:4]: This is JS syntax

File: server/controller/Songs/attribute.js (hunk 6 above)
Comment (khanhphd-student) [reply server/controller/Songs/attribute.js:1]: This is JS syntax

File: server/controller/Songs/attribute.js (hunk 7 above)
Comment (khanhphd-student) [reply server/controller/Songs/attribute.js:2]: I'll leave it there for future use

Comment (TrNgTien) [reply server/controller/Songs/attribute.js:2]: In our database that column have already generated automatically the ID, so we can delete that. Thanks @Dev01-vinamr-Ken

**Comment by TrNgTien** [comment]
Hey @Dev01-vinamr-Ken, I forgot I would like you return genres and artist of the song too, you will deal with that with BE. 
So API for songs is 
{
      songID: " ", //maybe we still need this. Sorry for this inconvenient
      songname : " " 
      uploadDate : " " ,
      numberOfLikes : " " ,
      duration : " ",
      genre : " ",
      artist : " "
      likeBy: [ userID, .... ];
}
And one more thing, we should return that song is liked by who! @Dev01-vinamr-Ken 
Sorry about this inconvenient.

File: server/controller/Songs/attribute.js (hunk 7 above)
Comment (khanhphd-student) [reply server/controller/Songs/attribute.js:2]: No

Comment (TrNgTien) [reply server/controller/Songs/attribute.js:2]: ok no problem with that

**Pull request merged**